File: CS/WindowsFormsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraBars.Ribbon;

namespace WindowsFormsApplication1 {
    public partial class Form1 : RibbonForm {
        public Form1() {
            InitializeComponent();
        }

        private void barButtonItem1_ItemClick(object sender, DevExpress.XtraBars.ItemClickEventArgs e) {
            DevExpress.Snap.SnapDocumentServer snapServer = new DevExpress.Snap.SnapDocumentServer();

            List<string> dataSourceTemplatesNames = new List<string>();
            dataSourceTemplatesNames.Add("nwindDataSourceTemplate.snx");
            dataSourceTemplatesNames.Add("carsDataBaseTemplate.snx");

            foreach(string dataSourceTemplate in dataSourceTemplatesNames) {
                using(System.IO.FileStream fs = new System.IO.FileStream(dataSourceTemplate, System.IO.FileMode.Open)) {
                    snapServer.LoadDocument(fs, DevExpress.Snap.Core.API.SnapDocumentFormat.Snap);

                    for(int i = 0; i < snapServer.Document.DataSources.Count; i++) {
                        if(snapServer.Document.DataSources[i].DataSource != null) {
                            snapControl1.Document.DataSources.Add(snapServer.Document.DataSources[i].DataSourceName, snapServer.Document.DataSources[i].DataSource);
                        }
                    }
                }                
            }            
        }
    }
}

File: VB/WindowsFormsApplication1/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraBars.Ribbon

Namespace WindowsFormsApplication1
	Partial Public Class Form1
		Inherits RibbonForm
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub barButtonItem1_ItemClick(ByVal sender As Object, ByVal e As DevExpress.XtraBars.ItemClickEventArgs) Handles barButtonItem1.ItemClick
			Dim snapServer As New DevExpress.Snap.SnapDocumentServer()

			Dim dataSourceTemplatesNames As New List(Of String)()
			dataSourceTemplatesNames.Add("nwindDataSourceTemplate.snx")
			dataSourceTemplatesNames.Add("carsDataBaseTemplate.snx")

			For Each dataSourceTemplate As String In dataSourceTemplatesNames
				Using fs As New System.IO.FileStream(dataSourceTemplate, System.IO.FileMode.Open)
					snapServer.LoadDocument(fs, DevExpress.Snap.Core.API.SnapDocumentFormat.Snap)

					For i As Integer = 0 To snapServer.Document.DataSources.Count - 1
						If snapServer.Document.DataSources(i).DataSource IsNot Nothing Then
							snapControl1.Document.DataSources.Add(snapServer.Document.DataSources(i).DataSourceName, snapServer.Document.DataSources(i).DataSource)
						End If
					Next i
				End Using
			Next dataSourceTemplate
		End Sub
	End Class
End Namespace
